lib/puma/queue_close.rb

```diff
@@ -1,25 +1,25 @@
-# Queue#close was added in Ruby 2.3.
-# Add a simple implementation for earlier Ruby versions.
-unless Queue.instance_methods.include?(:close)
-  class ClosedQueueError < StandardError; end
-  module Puma
-    module QueueClose
-      def initialize
-        @closed = false
-        super
-      end
-      def close
-        @closed = true
-      end
-      def closed?
-        @closed
-      end
-      def push(object)
-        raise ClosedQueueError if @closed
-        super
-      end
-      alias << push
+class ClosedQueueError < StandardError; end
+module Puma
+
+  # Queue#close was added in Ruby 2.3.
+  # Add a simple implementation for earlier Ruby versions.
+  #
+  module QueueClose
+    def initialize
+      @closed = false
+      super
     end
-    Queue.prepend QueueClose
+    def close
+      @closed = true
+    end
+    def closed?
+      @closed
+    end
+    def push(object)
+      raise ClosedQueueError if @closed
+      super
+    end
+    alias << push
   end
+  ::Queue.prepend QueueClose
```

Comment:
This comment isn't really 100% related to your changes @MSP-Greg , but I'm wondering if it strikes you as weird that Puma essentially monkey patches a class provided by the stdlib.

It seems like it'd be safer for us to just subclass Queue into a new puma-specific Queue class that provided the appropriate overrides for `close`, `closed?`, and `push`.

Reply:
I believe this is only used on Ruby 2.2?

A while ago I suggested continuing support of 2.2 & 2.3, but for Puma 6.0, maybe it's time...

Reply:
Oh yeah that makes sense. Might as well keep the monkey patch in the code base until we remove Ruby 2.2 support. 

Ruby 2.2 was EOL'd on March 31, 2018. Ruby 2.3 was EOL'd on March 31, 2019.